R/tm_a_regression.R

```diff
@@ -826,29 +813,30 @@ srv_a_regression <- function(id,
             env = list(
               plot = Reduce(function(x, y) call("+", x, y), c(plot, parsed_ggplot2_args))
             )
-          )
+          ),
+          name = "plot_6_call"
         )
       }
 
-      if (input_type == "Response vs Regressor") {
+      quosure <- if (input_type == "Response vs Regressor") {
         plot_type_0()
       } else {
-        plot_base()
+        plot_base_q <- plot_base()
         switch(input_type,
-          "Residuals vs Fitted" = plot_type_1(),
-          "Normal Q-Q" = plot_type_2(),
-          "Scale-Location" = plot_type_3(),
-          "Cook's distance" = plot_type_4(),
-          "Residuals vs Leverage" = plot_type_5(),
-          "Cook's dist vs Leverage" = plot_type_6()
+          "Residuals vs Fitted" = plot_base_q %>% plot_type_1(),
+          "Normal Q-Q" = plot_base_q %>% plot_type_2(),
+          "Scale-Location" = plot_base_q %>% plot_type_3(),
+          "Cook's distance" = plot_base_q %>% plot_type_4(),
+          "Residuals vs Leverage" = plot_base_q %>% plot_type_5(),
+          "Cook's dist vs Leverage" = plot_base_q %>% plot_type_6()
```

Comment:
There is a reactivity issue in this module when setting a filter:
1) Add filter `SEX` in the filter panel
2) Switch to any plot type
3) deselect and select `M` and `F` --> plot not updated
4) if you change the plot type, reactivity is triggered

Reply:
The problem is here lines 840, as output_q failed so the value could not be taken and the shiny error is not transferred too:

```
fitted <- reactive(output_q()[["fit"]])
plot_r <- reactive(output_q()[["g"]])
```

Reply:
resolved on the meeting after todays daily